src/server.js

```diff
@@ -88,14 +154,113 @@ export class Server {
       URI(this.serverURL)
         .segment('transactions')
         .toString(),
       `tx=${tx}`,
       { timeout: SUBMIT_TRANSACTION_TIMEOUT }
     )
-      .then((response) => response.data)
+      .then((response) => {
+        if (!response.data.result_xdr) {
+          return response.data;
+        }
+
+        const responseXDR = xdr.TransactionResult.fromXDR(
+          response.data.result_xdr,
+          'base64'
+        );
+
+        const results = responseXDR.result().value();
+
+        let offerResults;
+        let hasManageOffer;
+
+        if (results.length) {
+          offerResults = results
+            .map((result, i) => {
+              if (typeof result.value().value().success !== 'function') {
+                return null;
+              }
+
+              hasManageOffer = true;
+
+              let amountBought = new BigNumber(0);
+              let amountSold = new BigNumber(0);
+
+              const offerSuccess = result
+                .value()
+                .value()
+                .success();
+
+              const offersClaimed = offerSuccess
+                .offersClaimed()
+                .map((offerClaimed) => {
+                  const claimedOfferAmountBought = new BigNumber(
+                    // amountBought is a js-xdr hyper
+                    offerClaimed.amountBought().toString()
+                  );
+                  const claimedOfferAmountSold = new BigNumber(
+                    // amountBought is a js-xdr hyper
+                    offerClaimed.amountSold().toString()
+                  );
+
+                  // This is an offer that was filled by the one just submitted.
+                  // So this offer has an _opposite_ bought/sold frame of ref
+                  // than from what we just submitted!
+                  // So add this claimed offer's bought to the SOLD count and vice v
+
+                  amountBought = amountBought.add(claimedOfferAmountSold);
+                  amountSold = amountSold.add(claimedOfferAmountBought);
+
+                  return {
+                    sellerId: offerClaimed
+                      .sellerId()
+                      .value()
+                      .toString('hex'),
+                    offerId: offerClaimed.offerId,
+                    assetSold: offerClaimed.assetSold().switch(),
+                    amountSold: claimedOfferAmountSold
+                      .div(STROOPS_IN_LUMEN)
+                      .toString(),
+                    assetBought: offerClaimed.assetBought().switch(),
+                    amountBought: claimedOfferAmountBought
+                      .div(STROOPS_IN_LUMEN)
+                      .toString()
+                  };
+                });
+
+              const effect = offerSuccess.offer().switch().name;
+
+              return {
+                offersClaimed,
+                effect,
+                index: i,
+                lastOfferId: offersClaimed.reduce(
```

Comment:
Is `lastOfferId` intended to be the `offerID` of the offer that was just created/updated (I think this is the case based on the description above)? If so, that `offerID` should come from the `offer` field which exists in `offerSuccess` if and only if `effect != manageOfferDeleted`. Every `offerID` that comes from `offersClaimed` will always have existed in the ledger prior to this operation.

While mentioning the `offer` field, I thought you were going to add something like `remainingAmount` which could be gotten from that `offer` field?

Reply:
Yeah this doesn't seem that useful, instead I'm going to attach information about the created or updated outstanding offer if it exists.